src/marqo/tensor_search/tensor_search.py

```diff
@@ -1906,9 +1947,15 @@ def get_query_vectors_from_jobs(
             merged_vector = np.mean(weighted_vectors, axis=0)
 
             if q.index.normalize_embeddings:
-                norm = np.linalg.norm(merged_vector, axis=-1, keepdims=True)
-                if norm > 0:
-                    merged_vector /= np.linalg.norm(merged_vector, axis=-1, keepdims=True)
+                try:
+                    merged_vector = normalize_vector(merged_vector)
+                except core_exceptions.ZeroMagnitudeVectorError as e:
+                    if q.index.parsed_marqo_version() >= MARQO_CUSTOM_VECTOR_NORMALIZATION_MINIMUM_VERSION:
```

Comment:
Looks like this will error out for <2.13 anyway so we don't need this. Please verify that vespa errors out for 0 mag query and angular/prenormalized angular metric

Reply:
[Adding [slack conversation](https://marqo-ai.slack.com/archives/C05K6BFQUTF/p1727680962155369?thread_ts=1727339033.064519&cid=C05K6BFQUTF)] I'm not getting an error using either angular / prenormalized-angular as a distance metric here. 

This is the result with angular
`{'hits': [{'my_custom_vector': 'custom content is here!!', '_id': 'custom_vector_doc', '_highlights': [{'my_custom_vector': 'custom content is here!!'}], '_score': 0.38898452964834274}], 'query': None, 'limit': 10, 'offset': 0, 'processingTimeMs': 126}
`
the test case was:
```
settings = {
    "treat_urls_and_pointers_as_images": True,
    "model": "ViT-B/32",
    "ann_parameters": {
        "spaceType": "angular",
        "parameters": {"efConstruction": 512, "m": 16},
    },
    "normalize_embeddings": True,
}
mq.create_index("my-seventh-index", **settings)

result = mq.index('my-seventh-index').add_documents(
                documents=[{
                    "_id": "custom_vector_doc",
                                "my_custom_vector": {
                                    "content": "custom content is here!!",
                                    "vector": example_vector_1  # size is 512
                                }
                }],
                device="cpu", tensor_fields=["my_custom_vector", "text_field"],
                mappings={
            "my_custom_vector": {
                "type": "custom_vector"
            }
                })

search_result = mq.index('my-seventh-index').search(
    context={"tensor": [{"vector": zero_vector, "weight": 1} ]
             }
)

```

Result with prenormalised-angular: 

```
settings = {
    "treat_urls_and_pointers_as_images": True,
    "model": "ViT-B/32",
    "normalize_embeddings": True,
    "ann_parameters": {
    "spaceType": "prenormalized-angular",
    "parameters": {
      "efConstruction": 512,
      "m": 16
    }
  },
}

mq.create_index("my-eighth-index", **settings)
result = mq.index('my-eighth-index').add_documents(
    documents=[{
        "_id": "custom_vector_doc",
        "my_custom_vector": {
            "content": "custom content is here!!",
            "vector": example_vector_1  # size is 512
        }
    }],
    device="cpu", tensor_fields=["my_custom_vector", "text_field"],
    mappings={
        "my_custom_vector": {
            "type": "custom_vector"
        }
    })

search_result = mq.index('my-eighth-index').search(
    context={"tensor": [{"vector": zero_vector, "weight": 1} ]
             }
)
print(search_result)


```
`{'hits': [{'my_custom_vector': 'custom content is here!!', '_id': 'custom_vector_doc', '_highlights': [{'my_custom_vector': 'custom content is here!!'}], '_score': 0.5}], 'query': None, 'limit': 10, 'offset': 0, 'processingTimeMs': 5643}`

Since none of these threw an error, I believe we should have a version check, let me know if you don't agree - I can change it in that case.

Reply:
I'm alright with the version check if this is the case. Summarizing: if merged vector has magnitude zero: if marqo version is below the minimum, just ignore normalization, same behavior as before. If the version is equal to/greater: error with 400

Reply:
After having an offline discussion we have reverted logic to throw ZeroMagnitudeVectorError from search API. We will instead do this as part of a 3.x release